_drafts/gdbundle.md

```diff
@@ -0,0 +1,526 @@
+---
+title: "gdbundle - GDB's Missing Plugin Manager"
+description: "Introducing gdbundle, a plugin manager for GDB and LLDB, which enables developers to easily install plugins through PyPi and pip."
+author: tyler
+image:
+---
+
+I started with embedded development at Pebble, the smart watch maker, where we
+were building a modern operating system on a device with 128 KB of RAM. It's
+also where I got up-close and personal with GDB and had my first encounter with
+GDB's Python API.
+
+One one of our bi-weekly hack-days, one of the firmware engineers wrote a script
+to pretty-print a summary of the heap for a device connected to GDB using its
+Python API ([similar to
+this]({% post_url 2019-07-02-automate-debugging-with-gdb-python-api %}#adding-a-custom-gdb-command-with-gdb-python)).
+It was **awesome**. But, using this script with GDB wasn't without problems. We
+slowly grew it into a collection of scripts and improved our workflows. A few
+notable improvements were:
+
+- Add more scripts for each of our critical modules in the firmware, such as
+  state machines, Bluetooth state, etc.
+- Instead of just printing the heap allocations, expose the heap through a
+  Python Class. e.g.
+  `heap = Heap(g_heap); total_size = sum([b.size] for b in heap.iter_blocks()])`.
+- [Wrapping the GDB
+  invocation]({% post_url 2019-08-27-building-a-cli-for-firmware-projects %})
+  and loading GDB Python scripts automatically so every developer on the team
+  had them loaded and ready to use at all times.
+
+This was all back in 2015 or so. As I've moved around to other companies and
+talked to many other embedded software developers, I've discovered that many of
+them have never thought to extend or automate GDB using scripts, and even more
+had not even heard of GDB's Python API. I'm saddened by this.
+
+I have a personal goal (and it's one of Memfault's missions) to empower embedded
+software developers with the tools they need to get their job done, and I'm
+excited to share with you something I've been thinking about and building
+towards for over 3 years.
+
+<!-- excerpt start -->
+
+GDB desperately needs a better way for developers to share scripts,
+user-interface improvements, and utilities with the broader public. This would
+enable building upon each other's work and debugging techniques, progressing the
+entire community and preventing developers from reinventing the wheel time and
+time again. GDB (and LLDB) needs a plugin manager, and I'd like to introduce to
```

Comment:
```suggestion
entire community and preventing developers from reinventing the wheel. GDB (and LLDB) needs a plugin manager, and I'd like to introduce to
```

Reinventing the wheel time and time again is a repetition, both parts of the sentence mostly mean the same thing :-)